### ISSUE TITLE
Failing to match two similar routes with different methods when overwriteRoutes is true

### ISSUE THREAD

**Issue by sepo-one** [wheresrhys/fetch-mock#422]
A believe this one should be passing:
```javascript
it('overwriting similar routes with different methods', async () => {
  fm.config.overwriteRoutes = true;

  fm.mock('http://it.at.there/', 200, { method: 'POST' })
    .mock('http://another/', 200)
    .mock('http://it.at.there/', 200, { method: 'GET' })
    .catch();

  await fm.fetchHandler('http://another/', { method: 'GET' });
  expect(fm.calls(true).length).to.equal(1);
});
```

So clashes are being found by identifier (`http://it.at.there/` in this case):
```
const clashes = this.routes.filter(getPropertyComparer(route, 'identifier'));
```

Then in all routes a search is done for particular "clash" based on method.
But methods are different.
This results an index to be `-1` in `FetchMock.addRoute` which is why `'http://another/'` route is removed being the last one.

At the very least, i would error out for negative index in this place.
Otherwise the logic is quite fragile. I've spent quite some time to debug.

**Comment by sepo-one** [comment]
Simple return does the fix:
```javascript
FetchMock.addRoute = function(uncompiledRoute) {
	// ...

	if (overwriteRoutes === true) {
		const index = this.routes.indexOf(clashes.find(methodsMatch));

		if (index === -1) return;

		this._uncompiledRoutes.splice(index, 1, uncompiledRoute);
		return this.routes.splice(index, 1, route);
	}
```

But it does not feel to be robust one.
I would expect a search for clashes to respect the method as well.
(And other potential options like `query`).

**Comment by sepo-one** [comment]
Actually the fix would be:
```javascript
if (index > -1) {
			this._uncompiledRoutes.splice(index, 1, uncompiledRoute);
			return this.routes.splice(index, 1, route);
		}
```

Here is the [PR](https://github.com/wheresrhys/fetch-mock/pull/423).
(It has some code inspections checks failing; will get to it later).

**Comment by wheresrhys** [comment]
Fixed in 7.3.3
